core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/dto/ResourceAccessDto.java

```diff
@@ -0,0 +1,76 @@
+package com.sequenceiq.cloudbreak.api.endpoint.v4.dto;
+
+import static com.sequenceiq.cloudbreak.util.NullUtil.throwIfNull;
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+
+import com.sequenceiq.cloudbreak.exception.BadRequestException;
+
+public class ResourceAccessDto {
+
+    public static final String NULL_DTO_EXCEPTION_MESSAGE = "AccessDto should not be null";
+
+    public static final String INVALID_RESOURCE_ACCESS_DTO_EXCEPTION_MESSAGE = "One and only one value of the crn and name should be filled!";
+
+    private final String name;
+
+    private final String crn;
+
+    protected ResourceAccessDto(String name, String crn) {
+        this.name = name;
+        this.crn = crn;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getCrn() {
+        return crn;
+    }
+
+    public static void validate(ResourceAccessDto dto) {
+        throwIfNull(dto, () -> new IllegalArgumentException(NULL_DTO_EXCEPTION_MESSAGE));
+        if (dto.isNotValid()) {
+            throw new BadRequestException(INVALID_RESOURCE_ACCESS_DTO_EXCEPTION_MESSAGE);
+        }
+    }
+
+    public boolean isNotValid() {
+        return !isValid();
+    }
+
+    public boolean isValid() {
+        return isNotEmpty(name) ^ isNotEmpty(crn);
+    }
+
+    public static class ResourceAccessDtoBuilder {
+
+        private String name;
+
+        private String crn;
+
+        public static ResourceAccessDtoBuilder aResourceAccessDtoBuilder() {
+            return new ResourceAccessDtoBuilder();
+        }
+
+        public ResourceAccessDtoBuilder withName(String name) {
+            if (isNotEmpty(name)) {
+                this.name = name;
+            }
+            return this;
+        }
+
+        public ResourceAccessDtoBuilder withCrn(String crn) {
+            if (isNotEmpty(crn)) {
+                this.crn = crn;
+            }
+            return this;
+        }
+
+        public ResourceAccessDto build() {
+            return new ResourceAccessDto(name, crn);
+        }
```

Comment:
I would add the valid() call here. Otherwise the valid call has to be made every time, that is error-prone.

Reply:
I think this in a different way. The object itself does not require any validation. It's valid since it does nothing with the containing data. Tha validation depends on what the developer want to do with an already existing dto, but on the other hand you've got some point on the error-proneness

Reply:
I agree partially: yes, the users of this class should decide what they do with the data. On the other hand, we do not want to use dto-s that have invalid data in them, our paradigm is to send an HTTP error code and error message back to the user whenever we have invalid data. Under this paradigm I think it does make sense to add the validate() to the build method.
There could be cases when we are happy to have invalid dto-s: in that case I would add another build method, withNoValidation, that would disable validation. But I think there is no use-case for that now.